>/home/circleci/project/src/server/game/Entities/Creature/CreatureGroups.cpp:293:120: fatal error: unused parameter 'movement' [-Wunused-parameter]
        MovementGenerator const* moveGen = member->GetMotionMaster()->GetMovementGenerator([](MovementGenerator const* movement)->bool { return true; }, MOTION_SLOT_DEFAULT);

--- a/src/server/game/Entities/Creature/CreatureGroups.cpp
+++ b/src/server/game/Entities/Creature/CreatureGroups.cpp
@@ -290,7 +290,7 @@ void CreatureGroup::LeaderStartedMoving()
         float angle = pair.second->FollowAngle + float(M_PI); // for some reason, someone thought it was a great idea to invert relativ angles...
         float dist = pair.second->FollowDist;
 
-        MovementGenerator const* moveGen = member->GetMotionMaster()->GetMovementGenerator([](MovementGenerator const* movement)->bool { return movement->GetMovementGeneratorType() != FORMATION_MOTION_TYPE; }, MOTION_SLOT_DEFAULT);
+        MovementGenerator const* moveGen = member->GetMotionMaster()->GetMovementGenerator([](MovementGenerator const* movement)->bool { return movement->GetMovementGeneratorType() == FORMATION_MOTION_TYPE; }, MOTION_SLOT_DEFAULT);
         if (!moveGen)
             member->GetMotionMaster()->MoveFormation(_leader, dist, angle, pair.second->LeaderWaypointIDs[0], pair.second->LeaderWaypointIDs[1]);
     }

--- a/src/server/game/Entities/Creature/CreatureGroups.cpp
+++ b/src/server/game/Entities/Creature/CreatureGroups.cpp
@@ -290,8 +290,8 @@ void CreatureGroup::LeaderStartedMoving()
         float angle = pair.second->FollowAngle + float(M_PI); // for some reason, someone thought it was a great idea to invert relativ angles...
         float dist = pair.second->FollowDist;
 
-        MovementGenerator const* moveGen = member->GetMotionMaster()->GetMovementGenerator([](MovementGenerator const* movement)->bool { return true; }, MOTION_SLOT_DEFAULT);
-        if (!moveGen || moveGen->GetMovementGeneratorType() != FORMATION_MOTION_TYPE)
+        MovementGenerator const* moveGen = member->GetMotionMaster()->GetMovementGenerator([](MovementGenerator const* movement)->bool { return movement->GetMovementGeneratorType() != FORMATION_MOTION_TYPE; }, MOTION_SLOT_DEFAULT);
+        if (!moveGen)
             member->GetMotionMaster()->MoveFormation(_leader, dist, angle, pair.second->LeaderWaypointIDs[0], pair.second->LeaderWaypointIDs[1]);
     }
 }

